FlowCrypt/Functionality/Services/Folders Services/LocalFoldersProvider.swift

```diff
@@ -11,27 +11,45 @@ import RealmSwift
 
 protocol LocalFoldersProviderType {
     func fetchFolders(for userEmail: String) -> [FolderViewModel]
-    func removeFolders(for userEmail: String)
-    func save(folders: [Folder], for user: User)
+    func removeFolders(for userEmail: String) throws
+    func save(folders: [Folder], for user: User) throws
 }
 
-struct LocalFoldersProvider: LocalFoldersProviderType {
-    private let folderCache: EncryptedCacheService<FolderRealmObject>
+final class LocalFoldersProvider {
+    private let encryptedStorage: EncryptedStorageType
+
+    private var storage: Realm {
+        encryptedStorage.storage
+    }
 
     init(encryptedStorage: EncryptedStorageType) {
-        self.folderCache = EncryptedCacheService(encryptedStorage: encryptedStorage)
+        self.encryptedStorage = encryptedStorage
     }
+}
 
+extension LocalFoldersProvider: LocalFoldersProviderType {
     func fetchFolders(for userEmail: String) -> [FolderViewModel] {
-        return folderCache.getAll(for: userEmail).compactMap(FolderViewModel.init)
+        storage.objects(FolderRealmObject.self).where {
+            $0.user.email == userEmail
+        }.compactMap(FolderViewModel.init)
     }
```

Comment:
I prefer to see explicit returns - maybe there is some linter for this? That would be nice.